mne/morph.py

```diff
@@ -0,0 +1,731 @@
+    warn : bool
+        If True, warn if not all vertices were used.
+    xhemi : bool
+        Morph across hemisphere.
+
+    Notes
+    -----
+    .. versionadded:: X.X.X
+
+    See Also
+    --------
+    X
+
+    """
+
+    def __init__(self, src, subject_from=None, subject_to='fsaverage',
+                 subjects_dir=None, niter_affine=(100, 100, 10),
+                 niter_sdr=(5, 5, 3), grid_spacing=(5., 5., 5.), smooth=None,
+                 warn=True, xhemi=False, verbose=False):
+
+        # it's impossible to use the class without passing this check, so it
+        # only needs to be checked here
+        if not check_version('nibabel', '') or not check_version('dipy', ''):
+            raise ImportError(
+                'NiBabel (Python) and DiPy (Python) must be correctly '
+                'installed and accessible from Python')
+
+        # Set attributes
+        self.kind = None
+        self.subject_from = subject_from
+        self.subject_to = subject_to
+        self.subjects_dir = subjects_dir
+        self.niter_affine = niter_affine
+        self.niter_sdr = niter_sdr
+        self.grid_spacing = grid_spacing
+        self.smooth = smooth
+        self.warn = warn
+        self.xhemi = xhemi
+        self.morph_data = None
+
+        if src is None:
+            return
+
+        if isinstance(src, string_types):
+            src = read_forward_solution(src)['src']
+
+        if isinstance(src, SourceSpaces):
+            self.kind = src.kind
+            self.src = src
+        else:
+            raise ValueError('src must be an instance of SourceSpaces or a '
+                             'path to a saved instance of SourceSpaces')
+
+        if subject_from is None:
+            try:
+                self.subject_from = src[0]['subject_his_id']
+            except KeyError:
+                raise KeyError('subject id in src is None. Please specify'
+                               'subject_from')
+        self.morph_data = _compute_morph_data(self, src, verbose=verbose)
+
+    def __call__(self, stc_from, verbose=None):
+        """Morph data.
+
+        Parameters
+        ----------
+        stc_from : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            The source estimate to morph.
+
+        verbose : bool | str | int | None
+            If not None, override default verbose level (see :func:`mne.
+            verbose` and :ref:`Logging documentation <tut_logging>` for more).
+
+        Returns
+        -------
+        stc_to : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            The morphed source estimate.
+        """
+        if stc_from.subject is not None \
+                and stc_from.subject != self.subject_from:
+            raise ValueError('stc_from.subject and '
+                             'morph.subject_from must match')
+
+        # if VolSourceEstimate update mri properties for '.as_volume'
+        if self.kind is 'volume':
```

Comment:
nopes. Do

```py
self.kind == 'volume'
```

see: https://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce/1504742#1504742